frontend/src/metabase/actions/components/ActionViz/ActionVizForm.tsx

```diff
@@ -21,32 +22,38 @@ import { shouldShowConfirmation } from "./utils";
 import { FormWrapper, FormTitle } from "./ActionForm.styled";
 
 interface ActionFormProps {
-  onSubmit: OnSubmitActionForm;
+  action: WritebackAction;
   dashcard: ActionDashboardCard;
-  settings: VisualizationSettings;
-  isSettings: boolean;
   dashboard: Dashboard;
-  missingParameters: WritebackParameter[];
+  missingParameters?: WritebackParameter[];
+  mappedParameters?: WritebackParameter[];
   dashcardParamValues: ParametersForActionExecution;
-  action: WritebackQueryAction;
+  settings: VisualizationSettings;
+  isSettings: boolean;
   shouldDisplayButton: boolean;
   isEditingDashcard: boolean;
+  onSubmit: OnSubmitActionForm;
 }
 
 function ActionVizForm({
-  onSubmit,
+  action,
   dashcard,
-  settings,
-  isSettings,
   dashboard,
-  missingParameters,
+  settings,
+  missingParameters = [],
+  mappedParameters = [],
   dashcardParamValues,
-  action,
+  isSettings,
   shouldDisplayButton,
   isEditingDashcard,
+  onSubmit,
 }: ActionFormProps) {
   const [showModal, setShowModal] = useState(false);
   const title = getFormTitle(action);
+  const { getCleanValues } = useActionForm({
+    action,
+    initialValues: dashcardParamValues,
+  });
```

Comment:
I have mixed feelings about adding `useActionForm` to `ActionVizForm`. Most of the time, it'd end up rendering the `ActionForm`, which also uses the hook.

The reason it's here (and the reason `useActionForm` exists at all, TBH) is to handle a corner case when all parameters are mapped to filters, and we just need to execute the action on a button click ([see line 66](https://github.com/metabase/metabase/blob/f9c9003ff12098b4096dbcac936759d420a85e69/frontend/src/metabase/actions/components/ActionViz/ActionVizForm.tsx#L66)). I didn't want to add another utility for this single use-case as we're trying not to spread the logic all over the place

Would appreciate more sane alternatives